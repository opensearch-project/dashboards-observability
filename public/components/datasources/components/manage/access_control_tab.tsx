/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import {
  EuiButton,
  EuiFlexGroup,
  EuiFlexItem,
  EuiSpacer,
  EuiText,
  EuiHorizontalRule,
} from '@elastic/eui';
import React, { useEffect, useState } from 'react';
import { EuiPanel } from '@elastic/eui';
import { ConnectionManagementCallout } from './connection_management_callout';
import { coreRefs } from '../../../../framework/core_refs';
import { QueryPermissionsConfiguration } from './query_permissions';
import { DATACONNECTIONS_BASE } from '../../../../../common/constants/shared';
import { SaveOrCancel } from '../save_or_cancel';
import { Role } from '../../../../../common/types/data_connections';

interface AccessControlTabProps {
  dataConnection: string;
  connector: string;
  properties: unknown;
  allowedRoles: string[];
}

export const AccessControlTab = (props: AccessControlTabProps) => {
  const [mode, setMode] = useState<'view' | 'edit'>('view');
  const [roles, setRoles] = useState<Role[]>([]);
  const [selectedQueryPermissionRoles, setSelectedQueryPermissionRoles] = useState<Role[]>(
    props.allowedRoles.map((role) => {
      return { label: role };
    })
  );
  const { http } = coreRefs;

  useEffect(() => {
    http!.get('/api/v1/configuration/roles').then((data) =>
      setRoles(
        Object.keys(data.data).map((key) => {
          return { label: key };
        })
      )
    );
  }, []);

  const AccessControlDetails = () => {
    return (
      <EuiFlexGroup>
        <EuiFlexItem>
          <EuiFlexGroup direction="column">
            <EuiFlexItem grow={false}>
              <EuiText className="overview-title">Query access</EuiText>
              <EuiText size="s" className="overview-content">
                {selectedQueryPermissionRoles.length ? `Restricted` : '-'}
              </EuiText>
            </EuiFlexItem>
          </EuiFlexGroup>
        </EuiFlexItem>
      </EuiFlexGroup>
    );
  };

  const EditAccessControlDetails = () => {
    return (
      <EuiFlexGroup direction="column">
        <QueryPermissionsConfiguration
          roles={roles}
          selectedRoles={selectedQueryPermissionRoles}
          setSelectedRoles={setSelectedQueryPermissionRoles}
          layout={'vertical'}
        />
      </EuiFlexGroup>
    );
  };

  const saveChanges = () => {
    http!.put(`${DATACONNECTIONS_BASE}`, {
      body: JSON.stringify({
        name: props.dataConnection,
        allowedRoles: selectedQueryPermissionRoles.map((role) => role.label),
        connector: props.connector,
        properties: props.properties,
      }),
    });
    setMode('view');
  };

  const AccessControlHeader = () => {
    return (
      <EuiFlexGroup direction="row">
        <EuiFlexItem>
          <EuiText size="m">
            <h2 className="panel-title">Access Control</h2>
            Control which OpenSearch users have access to this data source.
          </EuiText>
        </EuiFlexItem>

        <EuiFlexItem grow={false}>
          <EuiButton
            data-test-subj="createButton"
            onClick={() => setMode(mode === 'view' ? 'edit' : 'view')}
            fill={mode === 'view' ? true : false}
          >
            {mode === 'view' ? 'Edit' : 'Cancel'}
          </EuiButton>
        </EuiFlexItem>
      </EuiFlexGroup>
    );
  };

  return (
    <>
      <EuiSpacer />
      <ConnectionManagementCallout />
      <EuiSpacer />
      <EuiPanel>
        <AccessControlHeader />
        <EuiHorizontalRule />
        {mode === 'view' ? <AccessControlDetails /> : <EditAccessControlDetails />}
      </EuiPanel>
      <EuiSpacer />
      {mode === 'edit' && (
        <SaveOrCancel
          onCancel={() => {
            setMode('view');
          }}
          onSave={saveChanges}
        />
      )}
      <EuiSpacer />
    </>
  );
};
