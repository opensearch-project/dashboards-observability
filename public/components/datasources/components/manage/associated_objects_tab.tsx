/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React, { useEffect, useState } from 'react';
import {
  EuiInMemoryTable,
  EuiLink,
  EuiPanel,
  EuiFlexGroup,
  EuiFlexItem,
  EuiText,
  EuiHorizontalRule,
  EuiButton,
  EuiSpacer,
  EuiEmptyPrompt,
} from '@elastic/eui';
import { AssociatedObject } from 'common/types/data_connections';
import { AccelerationsRecommendationCallout } from './accelerations_recommendation_callout';

interface AssociatedObjectsTabProps {
  associatedObjects: AssociatedObject[];
}

export const AssociatedObjectsTab: React.FC<AssociatedObjectsTabProps> = ({
  associatedObjects,
}) => {
  const [lastUpdated, setLastUpdated] = useState('');

  // TODO: FINISH THE REFRESH LOGIC
  const fetchAssociatedObjects = async () => {
    // Placeholder for data fetching logic
    // After fetching data:
    // setAssociatedObjects(fetchedData);
    const now = new Date();
    setLastUpdated(now.toUTCString()); // Update last updated time
  };

  useEffect(() => {
    fetchAssociatedObjects();
  }, []);

  const AssociatedObjectsHeader = () => {
    return (
      <EuiFlexGroup direction="row">
        <EuiFlexItem>
          <EuiText size="m">
            <h2 className="panel-title">Associated objects</h2>
            Manage objects associated with this data sources.
          </EuiText>
        </EuiFlexItem>
        <EuiFlexItem grow={false}>
          <div style={{ textAlign: 'right' }}>
            <EuiText color="subdued" style={{ fontSize: 'small' }}>
              Last updated at:
            </EuiText>
            <EuiText color="subdued" style={{ fontSize: 'small' }}>
              {lastUpdated}
            </EuiText>
          </div>
        </EuiFlexItem>
        <EuiFlexItem grow={false}>
          <EuiButton
            data-test-subj="freshButton"
            iconType="refresh"
            onClick={fetchAssociatedObjects}
          >
            Refresh
          </EuiButton>
        </EuiFlexItem>
      </EuiFlexGroup>
    );
  };

  const noDataMessage = (
    <EuiEmptyPrompt
      title={<h2>You have no associated objects</h2>}
      body={<p>Add or config tables from your data source or use Query Workbench.</p>}
      actions={
        <EuiButton
          color="primary"
          fill
          onClick={() => window.open('https://example.com', '_blank')}
          iconType="popout"
          iconSide="left"
        >
          Query Workbench
        </EuiButton>
      }
    />
  );

  const columns = [
    {
      field: 'name',
      name: 'Name',
      sortable: true,
      'data-test-subj': 'nameCell',
      render: (name: string) => (
        <EuiLink href="https://example.com" target="_blank">
          {name}
        </EuiLink>
      ),
    },
    {
      field: 'database',
      name: 'Database',
      truncateText: true,
      render: (database: string) => (
        <EuiLink href="https://example.com" target="_blank">
          {database}
        </EuiLink>
      ),
    },
    {
      field: 'type',
      name: 'Type',
      sortable: true,
    },
    {
      field: 'createdByIntegration',
      name: 'Created by Integration',
      sortable: true,
    },
    {
      field: 'accelerations',
      name: 'Accelerations',
      sortable: true,
    },
    {
      name: 'Actions',
      actions: [
        {
          name: 'Edit',
          description: 'Edit this object',
          type: 'icon',
          icon: 'discoverApp',
          onClick: (item: AssociatedObject) => console.log('Edit', item),
        },
        {
          name: 'Delete',
          description: 'Delete this object',
          type: 'icon',
          icon: 'bolt',
          onClick: (item: AssociatedObject) => console.log('Delete', item),
        },
      ],
    },
  ];

  const search = {
    box: {
      incremental: true,
      schema: {
        fields: { name: { type: 'string' }, database: { type: 'string' } },
      },
    },
  };

  const pagination = {
    initialPageSize: 10,
    pageSizeOptions: [10, 25, 50],
  };

  const sorting = {
    sort: {
      field: 'name',
      direction: 'asc',
    },
  };

  return (
    <>
      <EuiSpacer />
      <EuiPanel>
        <AssociatedObjectsHeader />
        <EuiHorizontalRule />
        <AccelerationsRecommendationCallout />
        <EuiSpacer />
        {associatedObjects.length > 0 ? (
          <EuiInMemoryTable
            items={associatedObjects}
            columns={columns}
            search={search}
            pagination={pagination}
            sorting={sorting}
            noItemsMessage={associatedObjects.length === 0 ? noDataMessage : undefined}
          />
        ) : (
          noDataMessage
        )}
      </EuiPanel>
      <EuiSpacer />
    </>
  );
};
