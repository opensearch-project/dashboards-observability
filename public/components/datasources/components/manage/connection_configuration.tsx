/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import {
  EuiFieldPassword,
  EuiFieldText,
  EuiFlexGroup,
  EuiFlexItem,
  EuiFormRow,
  EuiSelect,
  EuiSpacer,
  EuiText,
  EuiTextArea,
} from '@elastic/eui';
import { EuiSelectOption } from '@elastic/eui/src/components/form/select';
import React, { useState } from 'react';

interface ConnectionConfigurationProps {
  connectionName: string;
  connectionDetails: string;
  onConnectionDetailsChange: (e: any) => void;
  authenticationOptions: EuiSelectOption[];
  setSelectedAuthenticationMethod: (authenticationMethod: EuiSelectOption) => void;
  selectedAuthenticationMethod: string;
}

export const ConnectionConfiguration = (props: ConnectionConfigurationProps) => {
  const {
    connectionName,
    connectionDetails,
    onConnectionDetailsChange,
    authenticationOptions,
    selectedAuthenticationMethod,
    setSelectedAuthenticationMethod,
  } = props;
  const [details, setDetails] = useState(connectionDetails);

  const [password, setPassword] = useState('');

  const NameRow = () => {
    return (
      <EuiFlexGroup direction="row">
        <EuiFlexItem>
          <EuiText className="overview-title">Data source name</EuiText>
          <EuiText size="s" className="overview-content">
            This is the name of the data source and how it will be referenced in OpenSearch
            Dashboards.
          </EuiText>
        </EuiFlexItem>
        <EuiFlexItem>
          <EuiFormRow label="Data source name">
            <EuiFieldText placeholder={connectionName} readOnly />
          </EuiFormRow>
        </EuiFlexItem>
      </EuiFlexGroup>
    );
  };

  const SparkEndpointRow = () => {
    return (
      <EuiFlexGroup direction="row">
        <EuiFlexItem>
          <EuiText className="overview-title">Spark endpoint URL</EuiText>
          <EuiText size="s" className="overview-content">
            {
              "The URL for your Spark cluster and where your data is. This is what OpenSearch will connect to. The endpoint URL can't be changed. If you'd like to use another endpoint create a new data source."
            }
          </EuiText>
        </EuiFlexItem>
        <EuiFlexItem>
          <EuiFormRow label="Spark endpoint URL">
            <EuiFieldText placeholder={'-'} readOnly />
          </EuiFormRow>
        </EuiFlexItem>
      </EuiFlexGroup>
    );
  };

  return (
    <EuiFlexItem>
      <NameRow />
      <EuiSpacer />
      <EuiFlexGroup direction="row">
        <EuiFlexItem>
          <EuiText className="overview-title">Description - optional</EuiText>
          <EuiText size="s" className="overview-content">
            Text that can help identify the data source or share additional details
          </EuiText>
        </EuiFlexItem>
        <EuiFlexItem>
          <EuiFormRow label="Description">
            <EuiTextArea
              placeholder="Placeholder text"
              aria-label="test"
              value={details}
              onChange={(e) => {
                setDetails(e.target.value);
              }}
              onBlur={onConnectionDetailsChange}
            />
          </EuiFormRow>
        </EuiFlexItem>
      </EuiFlexGroup>
      <EuiSpacer />
      <SparkEndpointRow />
      <EuiSpacer />
      <EuiFlexGroup direction="row">
        <EuiFlexItem>
          <EuiText className="overview-title">Authentication details</EuiText>
          <EuiText size="s" className="overview-content">
            This is information used to authenticate and create a data source with Spark.
          </EuiText>
        </EuiFlexItem>
        <EuiFlexItem>
          <EuiFormRow label="Authentication method">
            <EuiSelect
              options={authenticationOptions}
              value={selectedAuthenticationMethod}
              onChange={(e) => setSelectedAuthenticationMethod(e)}
            />
          </EuiFormRow>
          <EuiFormRow label="Username">
            <EuiFieldText placeholder={'Username placeholder'} />
          </EuiFormRow>
          <EuiFormRow label="Password">
            <EuiFieldPassword
              type="dual"
              value={password}
              onChange={(e) => setPassword(e.target.value)}
            />
          </EuiFormRow>
        </EuiFlexItem>
      </EuiFlexGroup>
    </EuiFlexItem>
  );
};
