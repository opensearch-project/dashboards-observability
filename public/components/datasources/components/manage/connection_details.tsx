/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { EuiFlexGroup, EuiFlexItem, EuiSpacer, EuiText, EuiHorizontalRule } from '@elastic/eui';
import React, { useState } from 'react';
import { EuiPanel } from '@elastic/eui';
import { ConnectionManagementCallout } from './connection_management_callout';
import { PrometheusProperties, S3GlueProperties } from './data_connection';
import { DatasourceType } from '../../../../../common/types/data_connections';

interface ConnectionDetailProps {
  dataConnection: string;
  connector: DatasourceType;
  description: string;
  properties: S3GlueProperties | PrometheusProperties;
}

export const ConnectionDetails = (props: ConnectionDetailProps) => {
  const { dataConnection, connector, description, properties } = props;

  const S3ConnectionConfigurationView = () => {
    return (
      <EuiFlexGroup>
        <EuiFlexItem>
          <EuiFlexGroup direction="column">
            <EuiFlexItem grow={false}>
              <EuiText className="overview-title">Data source name</EuiText>
              <EuiText size="s" className="overview-content">
                {dataConnection}
              </EuiText>
            </EuiFlexItem>
            <EuiFlexItem grow={false}>
              <EuiText className="overview-title">Data source description</EuiText>
              <EuiText size="s" className="overview-content">
                {description || '-'}
              </EuiText>
            </EuiFlexItem>
          </EuiFlexGroup>
        </EuiFlexItem>
        <EuiFlexItem>
          <EuiFlexGroup direction="column">
            <EuiFlexItem grow={false}>
              <EuiText className="overview-title">Index store region</EuiText>
              <EuiText size="s" className="overview-content">
                {(properties as S3GlueProperties)['glue.indexstore.opensearch.region'] || '-'}
              </EuiText>
            </EuiFlexItem>
            <EuiFlexItem grow={false}>
              <EuiText className="overview-title">Index store URI</EuiText>
              <EuiText size="s" className="overview-content">
                {(properties as S3GlueProperties)['glue.indexstore.opensearch.uri'] || '-'}
              </EuiText>
            </EuiFlexItem>
          </EuiFlexGroup>
        </EuiFlexItem>
      </EuiFlexGroup>
    );
  };

  const PrometheusConnectionConfigurationView = () => {
    return (
      <EuiFlexGroup>
        <EuiFlexItem>
          <EuiFlexGroup direction="column">
            <EuiFlexItem grow={false}>
              <EuiText className="overview-title">Data source name</EuiText>
              <EuiText size="s" className="overview-content">
                {dataConnection}
              </EuiText>
            </EuiFlexItem>
            <EuiFlexItem grow={false}>
              <EuiText className="overview-title">Data source description</EuiText>
              <EuiText size="s" className="overview-content">
                {description || '-'}
              </EuiText>
            </EuiFlexItem>
          </EuiFlexGroup>
        </EuiFlexItem>
        <EuiFlexItem>
          <EuiFlexGroup direction="column">
            <EuiFlexItem grow={false}>
              <EuiText className="overview-title">Prometheus URI</EuiText>
              <EuiText size="s" className="overview-content">
                {(properties as PrometheusProperties)['prometheus.uri'] || '-'}
              </EuiText>
            </EuiFlexItem>
          </EuiFlexGroup>
        </EuiFlexItem>
      </EuiFlexGroup>
    );
  };

  const ConnectionConfigurationHeader = () => {
    return (
      <EuiFlexGroup direction="row">
        <EuiFlexItem>
          <EuiText size="m">
            <h2 className="panel-title">Data source configurations</h2>
            Control configurations for your data source.
          </EuiText>
        </EuiFlexItem>
      </EuiFlexGroup>
    );
  };

  return (
    <>
      <EuiSpacer />
      <ConnectionManagementCallout />
      <EuiSpacer />
      <EuiPanel>
        <ConnectionConfigurationHeader />
        <EuiHorizontalRule />
        {connector === 'S3GLUE' ? (
          <S3ConnectionConfigurationView />
        ) : (
          <PrometheusConnectionConfigurationView />
        )}
      </EuiPanel>
      <EuiSpacer />
      <EuiSpacer />
    </>
  );
};
