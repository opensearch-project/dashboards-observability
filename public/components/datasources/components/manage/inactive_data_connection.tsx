/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { EuiButton, EuiCallOut } from '@elastic/eui';
import React from 'react';
import {
  DATACONNECTIONS_BASE,
  DATACONNECTIONS_UPDATE_STATUS,
  EDIT,
} from '../../../../../common/constants/shared';
import { coreRefs } from '../../../../framework/core_refs';
import { useToast } from '../../../common/toast';
import { DatasourceDetails } from './data_connection';

interface InactiveDataConnectionCalloutProps {
  datasourceDetails: DatasourceDetails;
  fetchSelectedDatasource: () => void;
}

export const InactiveDataConnectionCallout = ({
  datasourceDetails,
  fetchSelectedDatasource,
}: InactiveDataConnectionCalloutProps) => {
  const { setToast } = useToast();
  const { http } = coreRefs;
  const enableDataSource = () => {
    http!
      .post(`${DATACONNECTIONS_BASE}${EDIT}${DATACONNECTIONS_UPDATE_STATUS}`, {
        body: JSON.stringify({ name: datasourceDetails.name, status: 'active' }),
      })
      .then(() => {
        setToast(`Data connection ${datasourceDetails.name} enabled successfully`);
        fetchSelectedDatasource();
      })
      .catch((err) => {
        console.error(err);
        setToast(`Data connection ${datasourceDetails.name} could not be enabled.`, 'danger');
      });
  };

  return (
    <EuiCallOut title="This data source connection is inactive" color="warning" iconType="help">
      <p>
        Associated objects and accelerations are not available while this connection is inactive.
      </p>
      <EuiButton onClick={enableDataSource} color="warning">
        Enable connection
      </EuiButton>
    </EuiCallOut>
  );
};
