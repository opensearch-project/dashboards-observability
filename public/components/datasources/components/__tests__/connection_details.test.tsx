/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import {
  testPrometheusConnectionDetails,
  testS3ConnectionDetails,
} from '../../../../../test/datasources';
import { ConnectionDetails } from '../manage/connection_details';

describe('Connection Details test', () => {
  configure({ adapter: new Adapter() });

  it('Renders connection details for s3 datasource', async () => {
    const wrapper = mount(<ConnectionDetails {...testS3ConnectionDetails} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('Renders connection details for prometheus datasource', async () => {
    const wrapper = mount(<ConnectionDetails {...testPrometheusConnectionDetails} />);

    expect(wrapper).toMatchSnapshot();
  });
});
