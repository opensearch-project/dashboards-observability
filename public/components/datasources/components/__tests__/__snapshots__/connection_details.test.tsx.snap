// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Connection Details test Renders connection details for prometheus datasource 1`] = `
<ConnectionDetails
  connector="PROMETHEUS"
  dataConnection="prom"
  description=""
  properties={
    Object {
      "prometheus.uri": "localhost:9201",
    }
  }
>
  <EuiSpacer>
    <div
      className="euiSpacer euiSpacer--l"
    />
  </EuiSpacer>
  <ConnectionManagementCallout>
    <EuiCallOut
      iconType="iInCircle"
      title="Configurations may be managed elsewhere."
    >
      <div
        className="euiCallOut euiCallOut--primary"
      >
        <div
          className="euiCallOutHeader"
        >
          <EuiIcon
            aria-hidden="true"
            className="euiCallOutHeader__icon"
            color="inherit"
            size="m"
            type="iInCircle"
          >
            <EuiIconIInCircle
              aria-hidden={true}
              className="euiIcon euiIcon--medium euiIcon--inherit euiCallOutHeader__icon"
              focusable="false"
              role="img"
              style={null}
            >
              <svg
                aria-hidden={true}
                className="euiIcon euiIcon--medium euiIcon--inherit euiCallOutHeader__icon"
                focusable="false"
                height={16}
                role="img"
                style={null}
                viewBox="0 0 16 16"
                width={16}
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M7.5 11.508 7.468 8H6.25V7h2.401l.03 3.508H9.8v1H7.5Zm-.25-6.202a.83.83 0 0 1 .207-.577c.137-.153.334-.229.59-.229.256 0 .454.076.594.23.14.152.209.345.209.576 0 .228-.07.417-.21.568-.14.15-.337.226-.593.226-.256 0-.453-.075-.59-.226a.81.81 0 0 1-.207-.568ZM8 13A5 5 0 1 0 8 3a5 5 0 0 0 0 10Zm0 1A6 6 0 1 1 8 2a6 6 0 0 1 0 12Z"
                  fillRule="evenodd"
                />
              </svg>
            </EuiIconIInCircle>
          </EuiIcon>
          <span
            className="euiCallOutHeader__title"
          >
            Configurations may be managed elsewhere.
          </span>
        </div>
        <EuiText
          color="default"
          size="s"
        >
          <div
            className="euiText euiText--small"
          >
            <EuiTextColor
              color="default"
              component="div"
            >
              <div
                className="euiTextColor euiTextColor--default"
              >
                Access to data may be managed in other systems outside of OpenSearch. Check with your administrator for additional configurations.
              </div>
            </EuiTextColor>
          </div>
        </EuiText>
      </div>
    </EuiCallOut>
  </ConnectionManagementCallout>
  <EuiSpacer>
    <div
      className="euiSpacer euiSpacer--l"
    />
  </EuiSpacer>
  <EuiPanel>
    <div
      className="euiPanel euiPanel--paddingMedium euiPanel--borderRadiusMedium euiPanel--plain euiPanel--hasShadow"
    >
      <ConnectionConfigurationHeader>
        <EuiFlexGroup
          direction="row"
        >
          <div
            className="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--directionRow euiFlexGroup--responsive"
          >
            <EuiFlexItem>
              <div
                className="euiFlexItem"
              >
                <EuiText
                  size="m"
                >
                  <div
                    className="euiText euiText--medium"
                  >
                    <h2
                      className="panel-title"
                    >
                      Data source configurations
                    </h2>
                    Control configurations for your data source.
                  </div>
                </EuiText>
              </div>
            </EuiFlexItem>
          </div>
        </EuiFlexGroup>
      </ConnectionConfigurationHeader>
      <EuiHorizontalRule>
        <hr
          className="euiHorizontalRule euiHorizontalRule--full euiHorizontalRule--marginLarge"
        />
      </EuiHorizontalRule>
      <PrometheusConnectionConfigurationView>
        <EuiFlexGroup>
          <div
            className="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--directionRow euiFlexGroup--responsive"
          >
            <EuiFlexItem>
              <div
                className="euiFlexItem"
              >
                <EuiFlexGroup
                  direction="column"
                >
                  <div
                    className="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--directionColumn euiFlexGroup--responsive"
                  >
                    <EuiFlexItem
                      grow={false}
                    >
                      <div
                        className="euiFlexItem euiFlexItem--flexGrowZero"
                      >
                        <EuiText
                          className="overview-title"
                        >
                          <div
                            className="euiText euiText--medium overview-title"
                          >
                            Data source name
                          </div>
                        </EuiText>
                        <EuiText
                          className="overview-content"
                          size="s"
                        >
                          <div
                            className="euiText euiText--small overview-content"
                          >
                            prom
                          </div>
                        </EuiText>
                      </div>
                    </EuiFlexItem>
                    <EuiFlexItem
                      grow={false}
                    >
                      <div
                        className="euiFlexItem euiFlexItem--flexGrowZero"
                      >
                        <EuiText
                          className="overview-title"
                        >
                          <div
                            className="euiText euiText--medium overview-title"
                          >
                            Data source description
                          </div>
                        </EuiText>
                        <EuiText
                          className="overview-content"
                          size="s"
                        >
                          <div
                            className="euiText euiText--small overview-content"
                          >
                            -
                          </div>
                        </EuiText>
                      </div>
                    </EuiFlexItem>
                  </div>
                </EuiFlexGroup>
              </div>
            </EuiFlexItem>
            <EuiFlexItem>
              <div
                className="euiFlexItem"
              >
                <EuiFlexGroup
                  direction="column"
                >
                  <div
                    className="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--directionColumn euiFlexGroup--responsive"
                  >
                    <EuiFlexItem
                      grow={false}
                    >
                      <div
                        className="euiFlexItem euiFlexItem--flexGrowZero"
                      >
                        <EuiText
                          className="overview-title"
                        >
                          <div
                            className="euiText euiText--medium overview-title"
                          >
                            Prometheus URI
                          </div>
                        </EuiText>
                        <EuiText
                          className="overview-content"
                          size="s"
                        >
                          <div
                            className="euiText euiText--small overview-content"
                          >
                            localhost:9201
                          </div>
                        </EuiText>
                      </div>
                    </EuiFlexItem>
                  </div>
                </EuiFlexGroup>
              </div>
            </EuiFlexItem>
          </div>
        </EuiFlexGroup>
      </PrometheusConnectionConfigurationView>
    </div>
  </EuiPanel>
  <EuiSpacer>
    <div
      className="euiSpacer euiSpacer--l"
    />
  </EuiSpacer>
  <EuiSpacer>
    <div
      className="euiSpacer euiSpacer--l"
    />
  </EuiSpacer>
</ConnectionDetails>
`;

exports[`Connection Details test Renders connection details for s3 datasource 1`] = `
<ConnectionDetails
  connector="S3GLUE"
  dataConnection="ya"
  description=""
  properties={
    Object {
      "glue.indexstore.opensearch.region": "us-west-2",
      "glue.indexstore.opensearch.uri": "y",
    }
  }
>
  <EuiSpacer>
    <div
      className="euiSpacer euiSpacer--l"
    />
  </EuiSpacer>
  <ConnectionManagementCallout>
    <EuiCallOut
      iconType="iInCircle"
      title="Configurations may be managed elsewhere."
    >
      <div
        className="euiCallOut euiCallOut--primary"
      >
        <div
          className="euiCallOutHeader"
        >
          <EuiIcon
            aria-hidden="true"
            className="euiCallOutHeader__icon"
            color="inherit"
            size="m"
            type="iInCircle"
          >
            <EuiIconEmpty
              aria-hidden={true}
              className="euiIcon euiIcon--medium euiIcon--inherit euiIcon-isLoading euiCallOutHeader__icon"
              focusable="false"
              role="img"
              style={null}
            >
              <svg
                aria-hidden={true}
                className="euiIcon euiIcon--medium euiIcon--inherit euiIcon-isLoading euiCallOutHeader__icon"
                focusable="false"
                height={16}
                role="img"
                style={null}
                viewBox="0 0 16 16"
                width={16}
                xmlns="http://www.w3.org/2000/svg"
              />
            </EuiIconEmpty>
          </EuiIcon>
          <span
            className="euiCallOutHeader__title"
          >
            Configurations may be managed elsewhere.
          </span>
        </div>
        <EuiText
          color="default"
          size="s"
        >
          <div
            className="euiText euiText--small"
          >
            <EuiTextColor
              color="default"
              component="div"
            >
              <div
                className="euiTextColor euiTextColor--default"
              >
                Access to data may be managed in other systems outside of OpenSearch. Check with your administrator for additional configurations.
              </div>
            </EuiTextColor>
          </div>
        </EuiText>
      </div>
    </EuiCallOut>
  </ConnectionManagementCallout>
  <EuiSpacer>
    <div
      className="euiSpacer euiSpacer--l"
    />
  </EuiSpacer>
  <EuiPanel>
    <div
      className="euiPanel euiPanel--paddingMedium euiPanel--borderRadiusMedium euiPanel--plain euiPanel--hasShadow"
    >
      <ConnectionConfigurationHeader>
        <EuiFlexGroup
          direction="row"
        >
          <div
            className="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--directionRow euiFlexGroup--responsive"
          >
            <EuiFlexItem>
              <div
                className="euiFlexItem"
              >
                <EuiText
                  size="m"
                >
                  <div
                    className="euiText euiText--medium"
                  >
                    <h2
                      className="panel-title"
                    >
                      Data source configurations
                    </h2>
                    Control configurations for your data source.
                  </div>
                </EuiText>
              </div>
            </EuiFlexItem>
          </div>
        </EuiFlexGroup>
      </ConnectionConfigurationHeader>
      <EuiHorizontalRule>
        <hr
          className="euiHorizontalRule euiHorizontalRule--full euiHorizontalRule--marginLarge"
        />
      </EuiHorizontalRule>
      <S3ConnectionConfigurationView>
        <EuiFlexGroup>
          <div
            className="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--directionRow euiFlexGroup--responsive"
          >
            <EuiFlexItem>
              <div
                className="euiFlexItem"
              >
                <EuiFlexGroup
                  direction="column"
                >
                  <div
                    className="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--directionColumn euiFlexGroup--responsive"
                  >
                    <EuiFlexItem
                      grow={false}
                    >
                      <div
                        className="euiFlexItem euiFlexItem--flexGrowZero"
                      >
                        <EuiText
                          className="overview-title"
                        >
                          <div
                            className="euiText euiText--medium overview-title"
                          >
                            Data source name
                          </div>
                        </EuiText>
                        <EuiText
                          className="overview-content"
                          size="s"
                        >
                          <div
                            className="euiText euiText--small overview-content"
                          >
                            ya
                          </div>
                        </EuiText>
                      </div>
                    </EuiFlexItem>
                    <EuiFlexItem
                      grow={false}
                    >
                      <div
                        className="euiFlexItem euiFlexItem--flexGrowZero"
                      >
                        <EuiText
                          className="overview-title"
                        >
                          <div
                            className="euiText euiText--medium overview-title"
                          >
                            Data source description
                          </div>
                        </EuiText>
                        <EuiText
                          className="overview-content"
                          size="s"
                        >
                          <div
                            className="euiText euiText--small overview-content"
                          >
                            -
                          </div>
                        </EuiText>
                      </div>
                    </EuiFlexItem>
                  </div>
                </EuiFlexGroup>
              </div>
            </EuiFlexItem>
            <EuiFlexItem>
              <div
                className="euiFlexItem"
              >
                <EuiFlexGroup
                  direction="column"
                >
                  <div
                    className="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--directionColumn euiFlexGroup--responsive"
                  >
                    <EuiFlexItem
                      grow={false}
                    >
                      <div
                        className="euiFlexItem euiFlexItem--flexGrowZero"
                      >
                        <EuiText
                          className="overview-title"
                        >
                          <div
                            className="euiText euiText--medium overview-title"
                          >
                            Index store region
                          </div>
                        </EuiText>
                        <EuiText
                          className="overview-content"
                          size="s"
                        >
                          <div
                            className="euiText euiText--small overview-content"
                          >
                            us-west-2
                          </div>
                        </EuiText>
                      </div>
                    </EuiFlexItem>
                    <EuiFlexItem
                      grow={false}
                    >
                      <div
                        className="euiFlexItem euiFlexItem--flexGrowZero"
                      >
                        <EuiText
                          className="overview-title"
                        >
                          <div
                            className="euiText euiText--medium overview-title"
                          >
                            Index store URI
                          </div>
                        </EuiText>
                        <EuiText
                          className="overview-content"
                          size="s"
                        >
                          <div
                            className="euiText euiText--small overview-content"
                          >
                            y
                          </div>
                        </EuiText>
                      </div>
                    </EuiFlexItem>
                  </div>
                </EuiFlexGroup>
              </div>
            </EuiFlexItem>
          </div>
        </EuiFlexGroup>
      </S3ConnectionConfigurationView>
    </div>
  </EuiPanel>
  <EuiSpacer>
    <div
      className="euiSpacer euiSpacer--l"
    />
  </EuiSpacer>
  <EuiSpacer>
    <div
      className="euiSpacer euiSpacer--l"
    />
  </EuiSpacer>
</ConnectionDetails>
`;
